model/src/main/kotlin/utils/FindingsMatcher.kt

```diff
@@ -292,62 +291,52 @@ internal fun associateLicensesWithExceptions(license: SpdxExpression): SpdxExpre
     // If this is not a compound expression, there can be no stand-alone license exceptions with belonging licenses.
     if (license !is SpdxCompoundExpression) return license
 
-    // Exclusively operate on AND-only expressions without further nested expressions.
-    val hasOnlyAndOperator = license.operator == SpdxOperator.AND && "(" !in license.toString()
-    if (!hasOnlyAndOperator) {
-        val children = license.children.map { associateLicensesWithExceptions(it) }
-        return SpdxCompoundExpression(
-            license.operator,
-            children[0],
-            children[1],
-            *children.drop(2).toTypedArray()
-        )
+    // Only search for stand-alone exceptions as part of AND-expressions.
+    if (license.operator == SpdxOperator.OR) {
+        return license.children.map { associateLicensesWithExceptions(it) }.reduce(SpdxExpression::or)
```

Comment:
Note that using reduce this will create a nested tree of compound expressions. I was thinking about adding a normalization to the `SpdxCompoundExpression` class either as a function or within the constructor to automatically merge with child compound expressions that use the same constructor. But for now you should better call:
```kotlin
SpdxCompoundExpression(SpdxExpression.OR, license.children.map { associateLicensesWithExceptions(it) })
```

Reply:
I was hesitant about adding this normalization to the constructor because I think it would be good if `"a AND (b AND c)".toSpdx().toString == "a AND (b AND c)"`, even though "a AND b AND c" would be semantically correct as well.